mobile-app/app/screens/AppNavigator/screens/Dex/DexRemoveLiquidityV2.tsx

```diff
@@ -0,0 +1,434 @@
+import { PricesSectionV2 } from '@components/PricesSectionV2'
+
+type Props = StackScreenProps<DexParamList, 'RemoveLiquidity'>
+
+export function RemoveLiquidityScreenV2 (props: Props): JSX.Element {
+  const logger = useLogger()
+  const client = useWhaleApiClient()
+  const toast = useToast()
+  const [fee, setFee] = useState<BigNumber>(new BigNumber(0.0001))
+  const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
+  const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const { getTokenPrice } = useTokenPrice()
+  // this component state
+  const [tokenAAmount, setTokenAAmount] = useState<BigNumber>(new BigNumber(0))
+  const [tokenBAmount, setTokenBAmount] = useState<BigNumber>(new BigNumber(0))
+  const [valid, setValidity] = useState(false)
+  // ratio, before times 100
+  const [amount, setAmount] = useState<BigNumber>(new BigNumber(0)) // to construct tx
+  const navigation = useNavigation<NavigationProp<DexParamList>>()
+
+  // transaction card component UI
+  const [tokenTransactionCardStatus, setTokenTransactionCardStatus] = useState<'error' | 'active' | ' undefined'>()
+  const [hasError, setHasError] = useState(false)
+  const [isInputFocus, setIsInputFocus] = useState(false)
+  const [tokenToRemove, setTokenToRemove] = useState<string>('')
+  const [percentageType, setPercentageType] = useState<string | undefined>()
+  const [showToast, setShowToast] = useState(false)
+  const TOAST_DURATION = 2000
+
+  // breakdown summary state
+  const [hasInputAmount, setHasInputAmount] = useState(false)
+
+  // gather required data
+  const { pair, pairInfo } = props.route.params
+  const tokenA = useSelector((state: RootState) => tokenSelector(state.wallet, pair.tokenA.id))
+  const tokenB = useSelector((state: RootState) => tokenSelector(state.wallet, pair.tokenB.id))
+
+  useEffect(() => {
+    setTokenTransactionCardStatus(hasError ? 'error' : isInputFocus ? 'active' : undefined)
+  }, [hasError, isInputFocus])
+
+  const removeLiquidity = (): void => {
+    if (hasPendingJob || hasPendingBroadcastJob) {
+      return
+    }
+    navigation.navigate('RemoveLiquidityConfirmScreen', { amount, pair, pairInfo, tokenAAmount, tokenBAmount, fee, tokenA, tokenB })
+  }
+
+  useEffect(() => {
+    client.fee.estimate()
+      .then((f) => setFee(new BigNumber(f)))
+      .catch(logger.error)
+  }, [])
+
+  useEffect(() => {
+    setValidity(
+      new BigNumber(tokenToRemove).isGreaterThan(new BigNumber(0)) &&
+      new BigNumber(tokenToRemove).isLessThanOrEqualTo(new BigNumber(pairInfo.amount)) &&
+      !hasPendingJob &&
+      !hasPendingBroadcastJob
+    )
+  }, [tokenToRemove])
+
+  function onPercentagePress (amount: string, type: AmountButtonTypes): void {
+    setPercentageType(type)
+    setShowToast(true)
+  }
+
+  // show toast for selected percentage input
+  useEffect(() => {
+    if (showToast && percentageType !== undefined) {
```

Comment:
the show toast can keep up when tapping 25% and 50% fast, can refer here and use the code https://github.com/DeFiCh/wallet/pull/3243/files#diff-5697436f85eee2dd1351013fcc253e7a1d679152e0859e92041fb11269f4de8aR110